src/Sulu/Bundle/AdminBundle/Resources/js/containers/Navigation/Navigation.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+import React from 'react';
+import {Navigation as NavigationComponent} from '../../components';
+import Router from '../../services/Router/Router';
+import navigationRegistry from './registries/NavigationRegistry';
+import type {NavigationItem} from './types';
+
+type Props = {
+    router: Router,
+    onNavigate: (route: string) => void,
+};
+
+const SULU_CHANGELOG_URL = 'https://github.com/sulu/sulu/releases';
+
+export default class Navigation extends React.Component<Props> {
+    handleNavigationItemClick = (value: *) => {
+        if (value && typeof value === 'string') {
+            this.props.router.navigate(value);
+            this.props.onNavigate(value);
+        }
+    };
+
+    handleLogoutClick = () => {
+        // TODO: Logout user here.
+    };
+
+    handleProfileEditClick = () => {
+        // TODO: Open profile edit overlay here.
+    };
+
+    isItemActive = (navigationItem: NavigationItem) => {
+        const {router} = this.props;
+
+        return (navigationItem.mainRoute && router.route.name === navigationItem.mainRoute) ||
+            (navigationItem.childRoutes && navigationItem.childRoutes.includes(router.route.name));
+    };
+
+    render() {
+        const navigationItems = navigationRegistry.get();
+
+        return (
+            <NavigationComponent
+                title="Sulu" // TODO: Get this dynamically from server
+                username="Hikaru Sulu" // TODO: Get this data from logged in user
+                suluVersion="2.0.0-RC1" // TODO: Get this dynamically from server
+                suluVersionLink={SULU_CHANGELOG_URL}
+                onLogoutClick={this.handleLogoutClick}
+                onProfileClick={this.handleProfileEditClick}
+            >
+                {navigationItems.map((navigationItem: NavigationItem) => (
+                    <NavigationComponent.Item
+                        key={navigationItem.id}
+                        value={navigationItem.mainRoute ? navigationItem.mainRoute : navigationItem.id}
```

Comment:
You are wrong.

First: Haven't changed anything here, this code is just moved.
Second: We had discussion already, its because I'm using this value not only to navigate.

Reply:
That's not true. First, you have changed this code, because `navigationItem` and `mainRoute` didn't exist before. Second, whatever you put here will be passed to the `router.navigate` function, which will fail if we pass the id here.